oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -0,0 +1,343 @@
+
+   We use `Jest`_ and `React-Testing-Library`_ to test React components.
+   We are deprecating the `Enzyme`_ library.
+   New tests must not use Enzyme, and any repositories planning to move to React 18 or newer need to
+   replace Enzyme.
+
+#. **Target the latest standardised JavaScript version (ECMA-262)**
+
+   **Rationale**: edX JavaScript should be written consistent with the latest
+   ECMA-262 specification in order to ensure future support, the largest
+   community and the availability of modern features. Currently, the ECMA-262
+   edition 6 (ES6) is the edition with the largest support and should be used.
+   To support this syntax in older browsers, use `Babel`_. Babel may also be
+   configured to add syntax extensions widely adopted by the community of our
+   recommended framework (e.g., `JSX`_).
+
+   **Exception**: Much of edX's existing front end code is written conformant
+   to the edition of ECMA-262 released in 2009 (ES5). Files written in ES5
+   should be gradually converted to the newer standard as new development in
+   those feature areas requires.
+
+   **Note**: edX previously used CoffeeScript, but its use has now been
+   deprecated. Community interest in TypeScript has also grown, but it and
+   other languages that do not follow the ECMA-262 spec should not be used.
+
+#. **JavaScript code should follow the edX ESLint configuration**
+
+   **Rationale**: In order to standardize and enforce Open edX's JavaScript
+   coding style across multiple codebases, edX has published an ESLint
+   configuration that provides an enforceable specification. The
+   `edX ESLint Config`_ is made available as an npm package that can be
+   installed into any Open edX package.
+
+   **Exception**: The `edX ESLint Config for ES5`_ may be used only where ES5
+   is in use. Both configs may be used in the same codebase through the
+   use of `ESLint configuration cascading`_.
+
+#. **JavaScript libraries should be installed via npm**
+
+   **Rationale**: It is important that JavaScript libraries are kept
+   up-to-date, and one key component is to make it as simple as possible
+   to perform upgrades. Projects that use npm maintain all of their
+   library requirements in a single `package.json` file, which can be
+   easily changed as the versions change.
+
+#. **JavaScript projects should publish a package-lock file**
+
+   **Rationale**: To keep dependencies up-to-date and builds consistent, JavaScript
+   projects should allow patch and minor upgrades in their package.json file and
+   commit a package-lock.json file. The package-lock.json file will keep a full list
+   of dependencies and their versions, ensuring when built for deployments the
+   version of libraries are consistent. This follows the same pattern used in edX
+   Python code - see `OEP 18`_ for more information. For more information on
+   package-lock files see `Package Lock`_.
+
+#. **Keep dependencies up to date by using Greenkeeper**
```

Comment:
Shouldn't this recommend Renovate instead?  See https://openedx.atlassian.net/wiki/spaces/AC/pages/1841791377/Set+up+Renovate+to+Automate+JavaScript+Dependency+Updates#Notes-on-Alternatives for reasons we started to ditch Greenkeeper.  As with the Redux note above we could update it after initial release, but I'd rather not risk people trying to start switching to Greenkeeper after reading this.